lib/portage/tests/news/test_NewsItem.py

```diff
@@ -1,109 +1,347 @@
+                    ]
+                }
+            )
+            self.assertTrue(item.isValid())
+            self.assertFalse(
+                item.isRelevant(
+                    self.vardb, self.settings, profile_prefix + self.profile
+                ),
+                msg=f"Expected {item} to be irrelevant, but it was relevant!",
             )
-        finally:
-            os.unlink(item.path)
 
     def testDisplayIfInstalled(self):
-        tmpItem = self.fakeItem[:].replace(
-            "#Display-If-Installed:", f"Display-If-Installed: {'sys-apps/portage'}"
+        self.vardb.cpv_inject("sys-apps/portage-2.0", {"SLOT": "0"})
+
+        item = self._createNewsItem({"display_if_installed": ["sys-apps/portage"]})
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
         )
 
-        try:
-            item = self._processItem(tmpItem)
-            self.assertTrue(
-                item.isRelevant(self.vardb, self.settings, self.profile),
-                msg=f"Expected {tmpItem} to be relevant, but it was not!",
-            )
-        finally:
-            os.unlink(item.path)
+        # Test the negative case: a single Display-If-Installed listing
+        # a package we don't have.
+        item = self._createNewsItem(
+            {"display_if_installed": ["sys-apps/i-do-not-exist"]}
+        )
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What about several packages and we have none of them installed?
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "dev-util/pkgcheck",
+                    "dev-util/pkgdev",
+                    "sys-apps/pkgcore",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What about several packages and we have one of them installed?
+        self.vardb.cpv_inject("net-misc/openssh-9.2_p1", {"SLOT": "0"})
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "net-misc/openssh",
+                    "net-misc/dropbear",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
+        )
+
+        # What about several packages and we have all of them installed?
```

Comment:
For `Display-If-Installed` you check the case in which you have all the packages installed, but for `Display-If-Profile` and `Display-If-Keyword` you don't check that case, which seems a bit inconsistent. Is there are reason for that?

Reply:
I don't think we can do that for Display-If-Profile (one can't have multiple selected), or am I missing something?

Good point on keywords. We should also add some tests for multiple constraints (mixed DIF) to ensure they interact correctly.

Reply:
Yes, you are right, I missed that multiple profiles don't make sense.

Having mixed constraints tested sounds good.